builder/llpcBuilderImplImage.cpp

```diff
@@ -906,14 +930,26 @@ Value* BuilderImplImage::CreateImageLoad(
                             coords,
                             derivatives);
 
+    // The intrinsics insist on an FP data type, so we need to bitcast if we want an integer data type.
+    Type* pIntrinsicDataTy = pResultTy;
+    if (pTexelTy->isIntOrIntVectorTy())
+    {
+        pIntrinsicDataTy = ConvertTypeToFp(pTexelTy);
+        if (pTexelTy != pResultTy)
+        {
+            // TFE
+            pIntrinsicDataTy = StructType::get(pTexelTy->getContext(), { pTexelTy, getInt32Ty() });
+        }
+    }
+
     SmallVector<Value*, 16> args;
-    Instruction *pResult = nullptr;
+    Value *pResult = nullptr;
```

Comment:
Why do you need change the type from Instruction to Value? it seems it is always instruction in following code.

Reply:
CreateBitCast returns a Value. The only place it needs to be Instruction is where it is the input to CreateWaterfallLoop.